### PR TITLE
Refactor S3 helpers to lib/aws

### PR DESCRIPTION
**Pull request by tdy**
Subsumed by #2870 and #2863 

---

I needed these helpers, so refactoring for general reuse.

---

Converting to draft now that this is a code refactor instead of just location refactor.

* [x] fix dir trailing slash
* [ ] convert fully to async style

### CODE REVIEW

**Comment by nicknytko** [comment]
While we're refactoring stuff, can we convert the functions to be one of async/await or callback style only?  It seems we're currently mixing the two and I'm worried this will lead to more confusion in the future.

File: lib/aws.js [hunk 1]
```diff
@@ -35,3 +35,53 @@ module.exports.getS3RVERConfiguration = function() {
         endpoint: new AWS.Endpoint('http://localhost:5000'),
     });
 };
+
+module.exports.uploadToS3 = async function (filePath, isDirectory, bucket, S3FilePath, localPath, callback) {
```
Comment (mwest1066) [inline lib/aws.js:39]: Following up on @nicknytko’s comment, `async` functions generally should not take `callback` parameters. I don’t mind whether this is `async`-style or `callback`-style, but it really shouldn’t be both at once.

File: lib/aws.js [hunk 2]
```diff
@@ -35,3 +35,53 @@ module.exports.getS3RVERConfiguration = function() {
...
+    const s3 = new AWS.S3();
+
+    let body;
+    if (isDirectory) {
+        body = '';
+        S3FilePath += '/';
```
Comment (mwest1066) [inline lib/aws.js:45]: What if the file already ends in `/`?

File: lib/aws.js [hunk 3]
```diff
@@ -35,3 +35,53 @@ module.exports.getS3RVERConfiguration = function() {
...
+    } else {
+        try {
+            body = await fs.promises.readFile(filePath);
+        } catch (err) {
+            callback(null, [filePath, S3FilePath, err]);
```
Comment (mwest1066) [inline lib/aws.js:50]: I don’t like the way we return a length-3 array on error and a string “OK” on success. We should always return the same type. Also we should return errors normally as actual errors. We can capture the error in the caller if we want to suppress it.

File: lib/aws.js (hunk 2 above)
Comment (tdy) [reply lib/aws.js]: Changed to `s3FilePath += s3FilePath.endsWith('/') ? '' : '/';`

**Pull request closed without merging**